src/test/java/uk/ac/cam/cl/dtg/segue/api/AdminFacadeTest.java

```diff
@@ -0,0 +1,423 @@
+        RegisteredUserDTO currentUser = new RegisteredUserDTO();
+        List<Long> targetUsers = List.of(2L);
+
+        expect(userManager.getCurrentRegisteredUser(mockRequest)).andReturn(currentUser);
+        expect(isUserAnAdminOrEventManager(userManager, currentUser)).andReturn(false);
+
+        replay(userManager);
+
+        try (Response response = adminFacade.modifyUsersTeacherPendingStatus(mockRequest, false, targetUsers)) {
+          assertEquals(Response.Status.FORBIDDEN.getStatusCode(), response.getStatus());
+          assertEquals("You must be staff to access this endpoint.",
+              response.readEntity(SegueErrorResponse.class).getErrorMessage());
+        }
+
+        verify(userManager);
+      }
+
+      @Test
+      void modifyUsersTeacherPendingStatus_unknownTarget_returnsBadRequest()
+          throws NoUserLoggedInException, SegueDatabaseException, NoUserException {
+        HttpServletRequest mockRequest = replayMockServletRequest();
+        RegisteredUserDTO currentUser = new RegisteredUserDTO();
+        List<Long> targetUsers = List.of(2L);
+
+        expect(userManager.getCurrentRegisteredUser(mockRequest)).andReturn(currentUser);
+        expect(isUserAnAdminOrEventManager(userManager, currentUser)).andReturn(true);
+
+        expect(userManager.getUserDTOById(2L)).andThrow(new NoUserException("No user found with this ID!"));
+
+        replay(userManager);
+
+        try (Response response = adminFacade.modifyUsersTeacherPendingStatus(mockRequest, false, targetUsers)) {
+          assertEquals(Response.Status.BAD_REQUEST.getStatusCode(), response.getStatus());
+          assertEquals("One or more users could not be found: [2]. ",
+              response.readEntity(SegueErrorResponse.class).getErrorMessage());
+        }
+
+        verify(userManager);
+      }
+
+      @Test
+      void modifyUsersTeacherPendingStatus_failureForOneOfMultipleTargets_returnsBadRequest()
+          throws NoUserLoggedInException, SegueDatabaseException, NoUserException, ContentManagerException {
+        HttpServletRequest mockRequest = replayMockServletRequest();
+        RegisteredUserDTO currentUser = new RegisteredUserDTO();
+        List<Long> targetUsers = List.of(2L, 3L);
+
+        expect(userManager.getCurrentRegisteredUser(mockRequest)).andReturn(currentUser);
+        expect(isUserAnAdminOrEventManager(userManager, currentUser)).andReturn(true);
+
+        expect(userManager.getUserDTOById(2L)).andThrow(new NoUserException("No user found with this ID!"));
+
+        expect(userManager.getUserDTOById(3L)).andReturn(prepareTestUser(3L, true));
+        expect(userManager.updateTeacherPendingFlag(3L, false)).andReturn(prepareTestUser(3L, false));
+
+        expect(emailManager.getEmailTemplateDTO("teacher_declined")).andReturn(new EmailTemplateDTO());
+        emailManager.sendTemplatedEmailToUser(anyObject(RegisteredUserDTO.class), anyObject(EmailTemplateDTO.class),
```

Comment:
Could you explain how this work, please? I'm not familiar with `expectLastCall`. Is the reason why this line is not inside `expect` because it comes right before `expectLastCall`?

Reply:
Robert explained to me that EasyMock expect() can’t be used to mock void methods and that I should use expectLastCall() instead

Reply:
Thank you, that makes sense!